sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -29,12 +42,32 @@
  */
 @ServiceClient(builder = DigitalTwinsClientBuilder.class)
 public class DigitalTwinsAsyncClient {
-    private final ClientLogger logger = new ClientLogger(DigitalTwinsAsyncClient.class);
+    private static final ClientLogger logger = new ClientLogger(DigitalTwinsAsyncClient.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
     private final DigitalTwinsServiceVersion serviceVersion;
     private final AzureDigitalTwinsAPI protocolLayer;
 
     DigitalTwinsAsyncClient(HttpPipeline pipeline, DigitalTwinsServiceVersion serviceVersion, String host) {
-        this.protocolLayer = new AzureDigitalTwinsAPIBuilder().host(host).pipeline(pipeline).buildClient();
+        final SimpleModule stringModule = new SimpleModule("String Serializer");
+        stringModule.addSerializer(new StdSerializer<String>(String.class, false){
+            @Override public void serialize(String value, JsonGenerator gen, SerializerProvider provider)
+                throws IOException {
+                if (isJSONValid(value)) {
```

Comment:
I wonder how costly this evaluation is?

Reply:
I don't have any numbers to compare with, but it will get evaluated for each String that is passed to the request.

Reply:
I worry about high overhead for that. Can we not safely assume all strings are json objects and so just always return true?

Reply:
Unfortunately no, because for a json string, we want the raw value to be sent to service, without any additional serialization - additional quotes (since it is a string). However, for a regular string, we do not want the raw value, but the string value instead (with quotes).

If we do it globally, something like digtalTwinId would get sent over as `dt_id_1`, instead of `"dt_id_1"`.

Reply:
Since all strings go through this serializer, we cannot make this differentiation without doing the convert-to-json-tree check.